website/blog/2020-06-16-alluxio-datacaching.md

```diff
@@ -0,0 +1,243 @@
+follows:
+<br><br>
+*`<BASE_DIR>/LOCAL/1048576/<BUCKET>/<PAGE>`*
+<br><br>
+Here:<br>
+
+1. BASE_DIR is the root directory of the cache storage and is set by Presto configuration “cache.base-directory”.
+2. LOCAL means the cache storage type is LOCAL. Alluxio also supports rocksDB as the cache storage.
+3. 1048576: represents the 1MB block size.
+4. BUCKET represents a directory serving as buckets for various page files. They are created to make sure one single directory does not have too many 
+files which often leads to really bad performance. 
+5. PAGE represents the file named after the page id. In the presto the ID is the md5 hash of the filename. 
+
+### Thread Concurrency
+
+Each Presto worker keeps a set of threads, each executing different query tasks but sharing the same data cache. Thus this Alluxio data cache 
+is required to be highly concurrent across threads to deliver high throughput. Namely, this data cache allows multiple threads to fetch the 
+same page concurrently, while still ensuring thread-safety for evictions. 
+
+### Cache Recovery
+
+Alluxio local cache tries its best to reuse the cache data present in the local cache directory when a worker starts up (or restarts). 
+If the cache directory structure is compatible, it reuses the cache data. 
+
+
+### Monitoring
+
+Alluxio exports various JMX metrics while performing various caching related operations. System admins can also monitor the cache usage across the cluster easily.
+
+# Presto+Alluxio Benchmark
+We benchmarked with queries from one of our production clusters, which was shadowed to the test cluster.
+<br><br>
+**Query Count**: 17320
+<br>
+**Cluster size**: 600 nodes
+<br>
+**Max cache capacity per node**: 460GB
+<br>
+**Eviction policy**: LRU
+<br>
+**Cache data block size**: 1MB, meaning data is read, stored, and evicted in the 1 MB size.
+<br><br>
+
+**Query Execution time improvement (in milliseconds)**:
+
+![](/img/blog/2020-06-16-alluxio-datacaching/query_latency.jpg)
+
+As you can see, we observed significant improvements in the query latencies. 
+We observed 33% improvement in P50, 54% improvement in P75, and 48% improvement in P95.
+<br><br>
+
+**IO Savings**
+<br>
+Data Size read for master branch run: **582 T Bytes**
+<br>
+Data Size read for caching branch run: **251 T Bytes**
+<br>
+Savings in Scans: **57%**
+<br><br>
+
+
+    
+**Cache hit rate**:
+
+![](/img/blog/2020-06-16-alluxio-datacaching/cache_hitrate.jpg)
+
+Cache hit rate was pretty consistent and good during the experiment run. It remained mostly between 0.9 and 1. 
+There were a few dips that could be noticed, these can be the result of a new query scanning lots of new data. 
+We need to implement additional algorithms to prevent less frequent data blocks to get 
+cached over more frequent data.
+
+# How to use it?
+In order to use data caching the first thing we need to do is to enable soft affinity. 
```

Comment:
Is this in the official documentation? Link to that instead.

We don't want to be one of those databases that buries documentation in the blog ;-)

Reply:
I don't have one at the moment, may be I can create it later and update the blog.

Reply:
OK, this is a pretty high priority. It's bad for project health to not maintain the documentation. And yes I am guilty as well!

Reply:
I was about to add it sometime back. However, Alluxio showed interest in contributing to the Presto via adding doc PR. Its added in the Alluxio task list. I can ask them in our sync-up today. If not I will add the documentation soon.